extensions/amp-a4a/0.1/real-time-config-manager.js

```diff
@@ -0,0 +1,217 @@
+  // and send the RTC request.
+  (rtcConfig['urls'] || []).forEach(url =>
+    inflateAndSendRtc_(a4aElement, url, seenUrls, promiseArray,
+        rtcStartTime, customMacros,
+        rtcConfig['timeoutMillis'])
+  );
+  // For each vendor the publisher has specified, inflate the vendor
+  // url if it exists, and send the RTC request.
+  Object.keys(rtcConfig['vendors'] || []).forEach(vendor => {
+    const url = RTC_VENDORS[vendor.toLowerCase()];
+    if (!url) {
+      return logAndAddErrorResponse(promiseArray,
+          RTC_ERROR_ENUM.UNKNOWN_VENDOR, vendor);
+    }
+    // The ad network defined macros override vendor defined/pub specifed.
+    const macros = Object.assign(rtcConfig['vendors'][vendor], customMacros);
+    inflateAndSendRtc_(a4aElement, url, seenUrls, promiseArray, rtcStartTime,
+        macros, rtcConfig['timeoutMillis'],
+        vendor);
+  });
+  return Promise.all(promiseArray);
+}
+
+/**
+ * @param {!AMP.BaseElement} a4aElement
+ * @param {!string} url
+ * @param {!Object<string, boolean>} seenUrls
+ * @param {!Array<Promise<!rtcResponseDef>>} promiseArray
+ * @param {!number} rtcStartTime
+ * @param {!Object} macros
+ * @param {!number} timeoutMillis
+ * @param {string=} opt_vendor
+ * @private
+ */
+function inflateAndSendRtc_(a4aElement, url, seenUrls, promiseArray,
+                            rtcStartTime, macros, timeoutMillis,
+                            opt_vendor) {
+  const win = a4aElement.win;
+  const ampDoc = a4aElement.getAmpDoc();
+  if (Object.keys(seenUrls).length == MAX_RTC_CALLOUTS) {
+    return logAndAddErrorResponse(
+        promiseArray, RTC_ERROR_ENUM.MAX_CALLOUTS_EXCEEDED,
+        opt_vendor || url);
+  }
+  const urlReplacements = Services.urlReplacementsForDoc(ampDoc);
+  const whitelist = {};
+  Object.keys(macros || {}).forEach(key => whitelist[key] = true);
+  url = urlReplacements.expandSync(
+      url, macros, undefined, whitelist);
+  if (!isSecureUrl(url)) {
+    return logAndAddErrorResponse(promiseArray, RTC_ERROR_ENUM.INSECURE_URL,
+        opt_vendor || url);
+  }
+  if (seenUrls[url]) {
+    return logAndAddErrorResponse(promiseArray, RTC_ERROR_ENUM.DUPLICATE_URL,
+        opt_vendor || url);
+  }
+  seenUrls[url] = true;
+  promiseArray.push(sendRtcCallout_(
+      url, rtcStartTime, win, timeoutMillis, opt_vendor || url));
+}
+
+/**
+ * @param {!string} url
+ * @param {!number} rtcStartTime
+ * @param {!Window} win
+ * @param {!number} timeoutMillis
+ * @param {!string} callout
+ * @return {Promise<!rtcResponseDef>}
+ * @private
+ */
+function sendRtcCallout_(url, rtcStartTime, win, timeoutMillis, callout) {
+  /**
+   * Note: Timeout is enforced by timerFor, not the value of
+   *   rtcTime. There are situations where rtcTime could thus
+   *   end up being greater than timeoutMillis.
+   */
+  return Services.timerFor(win).timeoutPromise(
+      timeoutMillis,
+      Services.xhrFor(win).fetchJson(
+          url, {credentials: 'include'}).then(res => {
```

Comment:
There will be situations where we may want to put ampCors: false here to ensure the same request is generated for every site allowing the response to be cached.  For now, let's add:  NOTE(bradfrizzel): we could include ampCors:false allowing the request to be cached across sites but for now assume that is not a required feature.